test/test_bbm_bbm_1d.jl

```diff
@@ -1,305 +1,294 @@
-                            change_entropy=3.4175334942543323e-7)
-
-        @test_allocations(semi, sol, allocs=10_000)
+@testitem "bbm_bbm_1d_basic_reflecting" setup=[Setup, BBMBBMEquation1D, AdditionalImports] begin
+    @test_trixi_include(joinpath(EXAMPLES_DIR, "bbm_bbm_1d_basic_reflecting.jl"),
+                        tspan=(0.0, 1.0),
+                        l2=[0.00022223640611354037 7.626931465504548e-9 0.0],
+                        linf=[0.0002947437933196184 1.7383174233387422e-8 0.0],
+                        cons_error=[9.701410286113879e-10 0.5469460962472683 0.0],
+                        change_waterheight=9.701410286113879e-10,
+                        change_velocity=0.5469460962472683,
+                        change_entropy=132.10935771957952,
+                        atol=1e-10,
+                        atol_ints=1e-10) # to make CI pass
+
+    @test_allocations(semi, sol, allocs=1_500)
+
+    # test upwind operators
+    D1 = upwind_operators(Mattsson2017; derivative_order = 1,
+                          accuracy_order = accuracy_order, xmin = mesh.xmin,
+                          xmax = mesh.xmax,
+                          N = mesh.N)
+    D2 = sparse(D1.plus) * sparse(D1.minus)
+    solver = Solver(D1, D2)
+    semi = Semidiscretization(mesh, equations, initial_condition, solver,
+                              boundary_conditions = boundary_conditions,
+                              source_terms = source_terms)
+    ode = semidiscretize(semi, (0.0, 1.0))
+    sol = solve(ode, Tsit5(), abstol = 1e-7, reltol = 1e-7,
+                save_everystep = false, callback = callbacks, saveat = saveat)
+    atol = 1e-7 # to make CI pass
+    rtol = 1e-12
+    errs = errors(analysis_callback)
+    l2 = [0.002345799818043513 3.254313503127441e-8 0.0]
+    l2_measured = errs.l2_error[:, end]
+    for (l2_expected, l2_actual) in zip(l2, l2_measured)
+        @test isapprox(l2_expected, l2_actual, atol = atol, rtol = rtol)
+    end
+    linf = [0.05625950533062252 6.815531732318192e-7 0.0]
+    linf_measured = errs.linf_error[:, end]
+    for (linf_expected, linf_actual) in zip(linf, linf_measured)
+        @test isapprox(linf_expected, linf_actual, atol = atol, rtol = rtol)
     end
 
-    @trixi_testset "bbm_bbm_1d_well_balanced" begin
-        @test_trixi_include(joinpath(EXAMPLES_DIR, "bbm_bbm_1d_well_balanced.jl"),
-                            tspan=(0.0, 1.0),
-                            l2=[5.625847040853183e-15 2.7315317493074205e-15 0.0],
-                            linf=[5.995204332975845e-15 1.12866211315047e-14 0.0],
-                            cons_error=[8.881784197001252e-16 1.6076981933671723e-15 0.0],
-                            change_waterheight=-8.881784197001252e-16,
-                            change_velocity=-1.6076981933671723e-15,
-                            change_entropy=-3.1086244689504383e-15)
-
-        @test_allocations(semi, sol, allocs=10_000)
+    @test_allocations(semi, sol, allocs=2_000)
```

Comment:
I needed to increase the allocations from `1_500` to `2_000` for some reason to make the test pass (the number of allocations was `1_920`). I have no idea why this was necessary. The same holds for the upwind allocations of the next `@testitem`.

Reply:
Interesting. I also have no immediate idea. Could be caused by less effective constant propagation or something like this 🤷

Reply:
Ok, but I think it's fine. It's only marginal.